### DISCUSSION TITLE
babel-plugin-react-remove-properties and babel-plugin-transform-react-remove-prop-types equivalents?

### DISCUSSION THREAD

**Discussion by reintroducing** [swc-project/swc#4248] [Q&A] [answered]
I'm porting a few libraries from using Babel to SWC for compilation before publishing to npm and I'm wondering if there are any viable replacements for [babel-plugin-react-remove-properties](https://www.npmjs.com/package/babel-plugin-react-remove-properties) and [babel-plugin-transform-react-remove-prop-types](https://www.npmjs.com/package/babel-plugin-transform-react-remove-prop-types)? 

I use a good amount of `data-test` attributes in my library to help facilitate testing through Cypress and React Testing Library and this is a common pattern for those tools. It would be great to be able to remove these when SWC compiles the library for distribution. Similarly, I use PropTypes and I see that they are not removed during compilation either.

Is there any way to solve these two issues using SWC? I've tried to search for this topic multiple times over the last few days and could not find an answer so I finally broke down and wrote this :)

**Comment by kwonoj** [comment, marked as answer]
I recall I wrote a custom proof of concept plugin for one of these at my prev. work. It's under private repo and I do not have the code anymore though. I would say as same as babel `plugin` does, writing a new plugin (https://github.com/swc-project/swc/discussions/3540) would be ideal way to go.

**Reply by reintroducing** [reply]
Thank you, I appreciate the quick reply. Looks like this is something that is coming to SWC (ability to write custom transforms) or is already doable, but undoubtedly much more involved than I currently have the time to tackle. This answers my question (and confirms why I could not really find much info on it), however, in that it is not currently something that is implemented or readily available, even as a custom plugin/transform.
